src/Home.tsx

```diff
@@ -106,39 +106,29 @@ const Home = ({
 				writableStream?.seek(offset)
 			}
 
-			const indexFileStart = Math.floor(formData.spanStart / 128000);
-			const indexFileCount = Math.ceil((formData.spanEnd - formData.spanStart) / 128000) + indexFileStart;
-			for (let indexFile = indexFileStart; indexFile <= indexFileCount; indexFile += 1) {
+			for (let batchIndex = formData.spanStart; batchIndex <= formData.spanEnd; batchIndex += 1000) {
 				if (controllerStop.signal.aborted) throw new Error('Fetching aborted');
 				if (controllerPause.signal.aborted) throw new Error('Paused');
 
-				const indexData: Uint8Array | string = await fetchIndexFile(currentNet, indexFile);
-				if (typeof indexData === 'string') {
+				const objectsData: string[] | string = await fetchBatchBlocks(currentNet, batchIndex);
```

Comment:
What you need to do is:
 * fetch numbers/OIDs you know from some queue and GET/write contents
 * once you have less than ~500 blocks (half of the "batch") spawn this request _asynchronously_
 * in the result handler push new data to the queue
 * rinse, repeat

The core idea is to have SEARCH requests be done concurrently to GETs, otherwise each time your queue depletes you have a delay before getting to the next batch.